src/test/java/htsjdk/samtools/reference/SamLocusAndReferenceIteratorTest.java

```diff
@@ -1,26 +1,87 @@
 package htsjdk.samtools.reference;
 
 import htsjdk.HtsjdkTest;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMRecord;
 import htsjdk.samtools.SamReader;
 import htsjdk.samtools.SamReaderFactory;
-import htsjdk.samtools.util.*;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import htsjdk.samtools.util.OverlapDetector;
+import htsjdk.samtools.util.SamLocusIterator;
+import htsjdk.samtools.util.SequenceUtil;
+import htsjdk.variant.vcf.SamFileUtils;
 import org.testng.Assert;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.io.File;
+import java.io.IOException;
 
 public class SamLocusAndReferenceIteratorTest extends HtsjdkTest {
     static private final File TEST_DATA_DIR = new File("src/test/resources/htsjdk/samtools/reference");
 
-    @Test
-    public void testSamLocusAndReferenceIterator() {
+    final File referenceAssembly18_trimmed = new File(TEST_DATA_DIR, "Homo_sapiens_assembly18.trimmed.fasta");
+    final File referenceWithTrailingNewWhiteSpace = new File(TEST_DATA_DIR, "reference_with_trailing_whitespace.fasta");
 
-        final File reference = new File(TEST_DATA_DIR, "Homo_sapiens_assembly18.trimmed.fasta");
-        final File samFile = new File(TEST_DATA_DIR, "simpleSmallFile.sam");
-        final ReferenceSequenceFile referenceSequenceFile = new FastaSequenceFile(reference, false);
+    File simpleSmallFileCram;
+    File simpleSmallFileSam;
+    File simpleSmallFileBam;
+
+    File leadingInsertionCram;
+    File leadingInsertionSam;
+    File leadingInsertionBam;
+
+    @DataProvider
+    Object[][] testSamLocusAndReferenceIteratorData() {
+        return new Object[][]{
+                {simpleSmallFileSam, false},
+                {simpleSmallFileBam, false},
+                {simpleSmallFileCram, true}
+        };
+    }
+
+    @BeforeSuite
```

Comment:
Could we change this from `@BeforeSuite` to `@BeforeClass` ?  That will limit the chance of confusing things happening when this setup method is unexpectedly run before unrelated tests. (https://testng.org/doc/documentation-main.html#annotations)